include/os/linux/kernel/linux/blkdev_compat.h

```diff
@@ -394,7 +394,11 @@ vdev_lookup_bdev(const char *path, dev_t *dev)
 static inline void
 bio_set_op_attrs(struct bio *bio, unsigned rw, unsigned flags)
 {
+#if defined(HAVE_BIO_BI_OPF)
+	bio->bi_opf |= rw | flags;
```

Comment:
This should be `bio->bi_opf = rw | flags;`  to match mainline.

I have a strong suspicion that the existing `bio->bi_rw |= rw | flags;` is wrong and should be using `=` rather than `|=`.  A quick look at torvalds/linux@50bfcd0cbfd39f490a5d7f0319aa1f4a1f762e9e shows that in-tree filesystem code had been using `=` rather than `|=` before `bio_set_op_attrs()` was introduced. If you look in other places, you will see a bit of `|=` being used, but torvalds/linux@93c5bdf7ab71bbdae27f8f51fa175e06f000d69d suggests that had been a mistake.

Reply:
Ok thanks, I'll try it out with the change and report back. I noticed that too, but I'm not super familiar with this feature, so decided the safest route was to minimize the change and see if anyone else had a comment on it like this.

Reply:
I've made the changes and it seems to still be working, so modified the patch to include the recommendation. I left the old code for `bio->bi_rw` as it was though.

Reply:
Looking at it more closely, we allocate `struct bio` in vdev_disk_dio_alloc() with kmem_zalloc(). Then we call bio_set_op_attrs(). The end result there is the same whether |= or = is used, although = is better since it avoids an unnecessary memory read.